### ISSUE TITLE
skrollr only works after resize in Chrome

### ISSUE THREAD

**Issue by AndrasDF** [Prinzhorn/skrollr#614]
I'm starting a new project with skrollr: http://www.spectral.net/_kunden/ebv/history/
Unfortunately in Chrome skrollr is not initialized correctly - only after the Browser window is resized. 
It seems that getMaxScroll is 0 and only after the browser window is resized has the correct value. I tried to re-initialize skrollr after a timeout and refresh elements, but that doesn't help. I don't understand why, because in a previous projekt no problem occured in Chrome. 

Firefox, Safari and IE works fine. Im testing with Chrome Version 37.0.2062.124

With best regards, Andras

**Comment by BCushing** [comment]
@silvenon - that solution worked for me, I set the timeout to 1000 to ensure that all elements were loaded (I have a form that's loaded with JS and was probably messing up the height of the document).

**Comment by robhadfield** [comment]
I had this same issue and it was related to content loaded via ajax.

I use [PACE](http://github.hubspot.com/pace/docs/welcome/)  as a loading monitor and was able to call the skrollr init and options from my Page Loader function : 

    	Pace.on('done', function(){
		PL.skrollrInit();
	});

**Comment by Bit32** [comment]
Also experiencing this issue

**Comment by oldlam** [comment]
I also got the same issue,
now I solve it by calling init inside window.onload

**Comment by Julesman** [comment]
YESSS!!! window.onload solved the issue for me. Thanks.

window.onload = function() {
    var s = skrollr.init();
};

**Comment by smekosh** [comment]
Confirming that the window.onload solution works well. Thanks @oldlam!

**Comment by ekasur** [comment]
guys, I try @silvenon code, and I put it in my code it's not working. It's make my parallax offset.

I try @Julesman code also, it's not working. Nothing happen. Just to clarify, that is javascript code is'n it ? I put that outside of my jquery script.

I am so confuse what's the problem actualy ?
My parralax still not good when we load the page. But when I resize it, it's work.
Anybody  can give me some advice or solution?
Thanks

**Comment by robhadfield** [comment]
@ekasur - A link to your page or your code would be useful. Any console errors?

**Comment by baskinco** [comment]
I'm experiencing the same issue on mobile, even after adding the window.onload solution. However I have a bit of a different implementation that enables/disables skrollr when the screen size is changed. 
```
$(window.onload = function () {
  // initialize skrollr if the window width is large enough
  if ($(window).width() < 768) {
    var s = skrollr.init({
        render: function(data) {
            //Debugging - Log the current scroll position.
            // console.log(data.curTop);
        }
    });
  }
  // disable/enable skrollr if the window is resized above/below 767px wide
  $(window).on('resize', function () {
    if ($(window).width() >= 767) {
      skrollr.init().destroy(); // destroy if over 767
    }
    if ($(window).width() < 768) {
      skrollr.init(); // enable if under 768
    }
  });
});
```

**Comment by robhadfield** [comment]
Hey @baskinco - have you followed the mobile directions re [adding the `skrollr-body` element? ](https://github.com/Prinzhorn/skrollr#what-you-need-in-order-to-support-mobile-browsers)

**Comment by baskinco** [comment]
@robhadfield yes. all content is within the main div id="skrollr-body", and no fixed elements are within that div. Skrollr works, but sporadically. Sometimes elements don't transition at the positions they should  until _after_ a refresh of the page or even just moving the phone from portrait to landscape and then back again. Any insight you have you be greatly appreciated! Thanks.

**Comment by robhadfield** [comment]
What's the reason for disabling and reinitialising like you're doing? It seems counter productive.

From the logic you have there it's never being initialised until after a resize on a window size below 767 - why are you wrapping the initialisation in a size condition at all?

The only initial render is : `if ($(window).width() < 768) {` - it only falls into the second trap after resize. I would wrap your initialise etc into separate functions that you could fire...

eg: ( ignore syntax obv... )

```
var skrollrDesktopInit = function() {
    // skrollr init for desktop
};

var skrollrMobileInit = function() {
    // skrollr init for mobile
};

var resizeFunc = function()  {
    // If < 768 { skrollrMobileInit } 
    // else { skrollrDesktopInit }
};

// Fire on window resize
$(window).on('resize', function () {
    resizeFunc();
}

// Fire on first load
resizeFunc();

```

**Comment by baskinco** [comment]
@robhadfield I'm fairly new with javascript so my reasoning may be a little unstable :) I see your logic with separating everything into functions that fire when needed. One thing that may affect this solution is that I don't want skrollr to fire on desktop (over 768px) at all, so in my mind there doesn't need to be a skrollrDesktopInit. More like a skrollrDesktopDestroy. Thoughts?

**Comment by robhadfield** [comment]
Ah I see what you're doing now, you want to disable above 768 and also a condition in case it resizes bigger... ok...

Here's a verbose (although untested) example :

```
var skrollrStart = function() {

    var winW = $(window).width(),
        _skrollr = skrollr.get(); // See if skrollr is already initialised

    if ( winW <= 768 ) {
        if ( _skrollr ) {
            skrollr.refresh();
        } else {
            skrollr.init();
        }
    } else {
        if ( _skrollr ) { skrollr.destroy(); }
    }

}

// Fire initially
$(window).on('load', function () {
    skrollrStart();
}

// Fire on resize
$(window).on('resize', function () {
    skrollrStart();
}
```

**Comment by baskinco** [comment]
@robhadfield  tested, and slightly modified. Works like a charm. (see below)

And if you feel up for another challenge, I could use a set of eyes on this one as well ;) https://github.com/Prinzhorn/skrollr/issues/861

Thanks for all your help on this .. much appreciated. 

```
var skrollrStart = function() {
    var winW = $(window).width(),
        _skrollr = skrollr.get(); // See if skrollr is already initialised

    if ( winW <= 768 ) {
        if ( _skrollr ) {
            skrollr.refresh();
        } else {
            skrollr.init();
        }
    } else {
        if ( _skrollr ) {
          skrollr.init().destroy();
        }
    }
}

// Fire initially
$(window).on('load', function () {
    skrollrStart();
});

// Fire on resize
$(window).on('resize', function () {
    skrollrStart();
});
```

**Comment by robhadfield** [comment]
Awesome! I'm glad it worked, I really was working a blind so glad you managed to make sense of it :D

**Comment by fabianshamano** [comment]
Hi guys, I really hope you guys can help me I'm using scrollr for some really important projects and I've noticed 2 very fatal problems 
1. Anchor tags don't work, for instance if I wanna anchor a section of my site to my menu bar at the top it doesn't work at all 

2. Mobile scrolling of other elements is disabled. say I have a panel that has a set height and overflow auto, it will will work fine when testing on the pc however when tested on a mobile device the scroll bars appear but content cannot be scrolled. I believe scrollr takes preference over all the scrolling on the page making it close to useless for mobile application. 

I'm not a guru, still learning but I was able to confirm that these issues are caused by scrollr, may someone kindly help me please

**Comment by robhadfield** [comment]
Hi @fabianshamano - Firstly, if the project is 'really important' maybe don't rely on inexperienced developers and shouting at authors of free software... but I digress.

For menu anchors - please see here: https://github.com/Prinzhorn/skrollr-menu

I suspect your overflow issue is an IOS issue rather than a Skrollr issue? There is lots of content online about these issues, for example: https://codepen.io/chriscoyier/pen/nHEDj

Let us know how you get on.

**Comment by fabianshamano** [comment]
I did not shout at anyone in anyway only asked for assistance, perhaps my mistake was over explaining my predicament and no I'm not using IOS, but thanks anyway will take a look at the links

**Comment by robhadfield** [comment]
This isn't my repo but using phrases like "doesn't work at all" and "close to useless" will win you no friends.

**Comment by fabianshamano** [comment]
Mate you're clearly reading what I wrote the wrong way, I was wondering what would cause you to respond to me the way you did but now I think I see it. like I said my mistake was over explaining myself. The anchors don't work, that's not me criticising scrollr it's me stating my challenge that I need help with. 'Close to useless' I'm referring to my site on mobile devices as people won't be able to see the content if they can't scroll .

**Comment by robhadfield** [comment]
We're a community and drama over not being able to get it to work is not necessary. I wasn't that upset or I wouldn't have tried to help ;) 

Anyway - did you get it fixed?
